candidates/views/constituencies.py

```diff
@@ -306,8 +305,13 @@ def post(self, request, *args, **kwargs):
             with transaction.atomic():
                 post = get_object_or_404(Post, extra__slug=post_id)
                 lock = form.cleaned_data['lock']
-                post.extra.candidates_locked = lock
-                post.extra.save()
+                # FIXME check this exists
```

Comment:
Rather than leave this is a FIXME, I think it'd be better to create an issue to add code to check that there's exactly one postextraelection for each post / election associate to `check_constraints` and `candidates_check_for_inconsistent_data`.